sources/updater/updog/src/main.rs

```diff
@@ -437,6 +439,23 @@ fn output<T: Serialize>(json: bool, object: T, string: &str) -> Result<()> {
     Ok(())
 }
 
+fn initiate_reboot() -> Result<()> {
+    // Set up signal handler for termination signals
+    let signals = Signals::new(&[SIGINT, SIGTERM, SIGQUIT]).context(error::Signal)?;
```

Comment:
I think we should scope this to the signal(s) we actually get from the shutdown process.  That's just TERM, right?  (brupop isn't sending anything special?)  INT and QUIT are typically signals from humans, and if a human (or their code) is calling updog and they kill it because reboot hung or whatever, we don't want it to look successful.